### DISCUSSION TITLE
Generate PDF report from command line

### DISCUSSION THREAD

**Discussion by alexyz41** [GoogleChrome/lighthouse#12408] [General] [answered]
Hello,

is there any way to generate the report directly to PDF? Right now I am generating an HTML and converting it to PDF, but it would be nice if there is an option to generate the report as a PDF right away.

Thanks

**Comment by connorjclark** [comment, marked as answer]
Nope. Are you aware you can save as PDF via the print menu in Chrome? Or are you using the CLI?

**Reply by alexyz41** [reply]
Thanks for replying! Yes I am aware, but I need to do it for 200+ reports, so I need an automatic solution to generate the report directly as PDF. It is not feasible to do it one by one.

**Reply by connorjclark** [reply]
Agreed. You could use the Lighthouse CLI.

```
lighthouse https://www.example.com --output-path=stdout --output=html | somehtmltopdftool
```

Some possible tools are: https://unix.stackexchange.com/questions/533886/is-there-a-command-line-tool-for-converting-html-files-to-pdf

**Reply by alexyz41** [reply]
Thanks connorjclark! I will use your approach.

I was investigating these tools, but none of them did the job. PDFs were not well rendered. I finally got it with chrome headless. The command I used was:

`start chrome --enable-logging --headless --disable-gpu --print-to-pdf-no-header --use-system-default-printer --print-to-pdf=OUTPUT_FILE.pdf PATH_TO_HTML_FILE`

Thanks for your help!!

**Reply by connorjclark** [reply]
Good to know chrome can do the job, thanks for sharing the command!
